exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -792,7 +792,7 @@ private void setupOutputContainerSchema() {
         }
 
         // make sure to project field with children for children to show up in the schema
-        final MaterializedField projected = field.withType(outputType);
+        final MaterializedField projected = field.copy(outputType);
```

Comment:
The `withType` is clearer. With `copy`, we have to infer that this is "copy fields from the give field except for type, which we are changing." `withType` is similar to usages of this pattern elsewhere.

Reply:
`copy` refers to a copy constructor with a deep copy semantics. `with...` usually applies to a builder and that pattern is secondary here.

Reply:
In such case one of the `copy` should be named `copyWithNameAndType`. As parameters already provide a hint that `name` and/or `type` will be changed I don't see much need to use the method name for the same.